adapters/concert/concert.go

```diff
@@ -0,0 +1,165 @@
+package concert
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"net/http"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+const adapterVersion = "1.0.0"
+
+// Builder builds a new instance of the Concert adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	bidderImpExt, err := getBidderExt(request.Imp[0])
+	if err != nil {
+		return nil, []error{fmt.Errorf("get bidder ext: %v", err)}
+	}
+
+	requestJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	var requestMap map[string]interface{}
+	err = json.Unmarshal(requestJSON, &requestMap)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	if requestMap["ext"] == nil {
+		requestMap["ext"] = make(map[string]interface{})
+	}
+	requestMap["ext"].(map[string]interface{})["adapterVersion"] = adapterVersion
+	requestMap["ext"].(map[string]interface{})["partnerId"] = bidderImpExt.PartnerId
+
+	requestJSON, err = json.Marshal(requestMap)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+	headers.Add("Accept", "application/json")
+	requestData := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    requestJSON,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}
+
+	return []*adapters.RequestData{requestData}, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
```

Comment:
This is correct. But you could also use - 
```
if adapters.IsResponseStatusCodeNoContent(response) {
		return nil, nil
}

if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
		return nil, []error{err}
}
```

Reply:
@BrettBlox This doesn't seem to have been implemented.

Reply:
@gargcreation1992 Correct. Is this change a requirement? Your previous comment made it seem like a suggested alternative and that both options were valid.

Reply:
It's not a requirement but I would highly recommend making this change.

Reply:
@gargcreation1992 done.